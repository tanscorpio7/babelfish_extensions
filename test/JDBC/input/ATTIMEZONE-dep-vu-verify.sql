SELECT * FROM ATTIMEZONE_dep_vu_prepare_v1
GO

SELECT * FROM ATTIMEZONE_dep_vu_prepare_v2
GO

EXEC ATTIMEZONE_dep_vu_prepare_p1
GO

EXEC ATTIMEZONE_dep_vu_prepare_p2
GO

SELECT ATTIMEZONE_dep_vu_prepare_f1()
GO

SELECT ATTIMEZONE_dep_vu_prepare_f2()
GO

Select convert(datetime2,'2002-01-01 02:01:00.000 +00:00') AT TIME ZONE 'eastern standard time';
GO

Select convert(datetime2,'9999-12-31 15:59:59.000 +00:00') AT TIME ZONE 'Central Europe Standard Time';
GO

Select convert(datetimeoffset,'2002-01-01 02:01:00.000 +00:00') AT TIME ZONE 'eastern standard time';
GO

Select convert(datetimeoffset,'9999-12-31 15:59:59.000 +00:00') AT TIME ZONE 'Central Europe Standard Time';
GO

select set_config('timezone', 'Asia/Kolkata', false);
GO

Select convert(datetime2,'2002-01-01 02:01:00.000 +00:00') AT TIME ZONE 'eastern standard time';
GO

Select convert(datetime2,'9999-12-31 15:59:59.000 +00:00') AT TIME ZONE 'Central Europe Standard Time';
GO

Select convert(datetimeoffset,'2002-01-01 02:01:00.000 +00:00') AT TIME ZONE 'eastern standard time';
GO

Select convert(datetimeoffset,'9999-12-31 15:59:59.000 +00:00') AT TIME ZONE 'Central Europe Standard Time';
GO

select set_config('timezone', 'UTC', false);
GO
