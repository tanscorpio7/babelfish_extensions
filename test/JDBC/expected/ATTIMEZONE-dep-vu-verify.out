SELECT * FROM ATTIMEZONE_dep_vu_prepare_v1
GO
~~START~~
datetimeoffset
2002-01-01 02:01:00.0000000 +01:00
~~END~~


SELECT * FROM ATTIMEZONE_dep_vu_prepare_v2
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Argument data type varchar is invalid for argument 1 of AT TIME ZONE function.)~~


EXEC ATTIMEZONE_dep_vu_prepare_p1
GO
~~START~~
datetimeoffset
9999-12-31 16:59:59.0000000 +01:00
~~END~~


EXEC ATTIMEZONE_dep_vu_prepare_p2
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Argument data type varchar is invalid for argument 1 of AT TIME ZONE function.)~~


SELECT ATTIMEZONE_dep_vu_prepare_f1()
GO
~~START~~
datetimeoffset
2002-01-01 02:01:00.0000000 -05:00
~~END~~


SELECT ATTIMEZONE_dep_vu_prepare_f2()
GO
~~START~~
datetimeoffset
~~ERROR (Code: 33557097)~~

~~ERROR (Message: Argument data type varchar is invalid for argument 1 of AT TIME ZONE function.)~~


Select convert(datetime2,'2002-01-01 02:01:00.000 +00:00') AT TIME ZONE 'eastern standard time';
GO
~~START~~
datetimeoffset
2002-01-01 02:01:00.0000000 -05:00
~~END~~


Select convert(datetime2,'9999-12-31 15:59:59.000 +00:00') AT TIME ZONE 'Central Europe Standard Time';
GO
~~START~~
datetimeoffset
9999-12-31 15:59:59.0000000 +01:00
~~END~~


Select convert(datetimeoffset,'2002-01-01 02:01:00.000 +00:00') AT TIME ZONE 'eastern standard time';
GO
~~START~~
datetimeoffset
2001-12-31 21:01:00.0000000 -05:00
~~END~~


Select convert(datetimeoffset,'9999-12-31 15:59:59.000 +00:00') AT TIME ZONE 'Central Europe Standard Time';
GO
~~START~~
datetimeoffset
9999-12-31 16:59:59.0000000 +01:00
~~END~~


select set_config('timezone', 'Asia/Kolkata', false);
GO
~~START~~
text
Asia/Kolkata
~~END~~


Select convert(datetime2,'2002-01-01 02:01:00.000 +00:00') AT TIME ZONE 'eastern standard time';
GO
~~START~~
datetimeoffset
2002-01-01 02:01:00.0000000 -05:00
~~END~~


Select convert(datetime2,'9999-12-31 15:59:59.000 +00:00') AT TIME ZONE 'Central Europe Standard Time';
GO
~~START~~
datetimeoffset
9999-12-31 15:59:59.0000000 +01:00
~~END~~


Select convert(datetimeoffset,'2002-01-01 02:01:00.000 +00:00') AT TIME ZONE 'eastern standard time';
GO
~~START~~
datetimeoffset
2001-12-31 21:01:00.0000000 -05:00
~~END~~


Select convert(datetimeoffset,'9999-12-31 15:59:59.000 +00:00') AT TIME ZONE 'Central Europe Standard Time';
GO
~~START~~
datetimeoffset
9999-12-31 16:59:59.0000000 +01:00
~~END~~


select set_config('timezone', 'UTC', false);
GO
~~START~~
text
UTC
~~END~~

